client/src/pages/Contest/NewContestForm/NewContestForm.tsx

```diff
@@ -0,0 +1,220 @@
+import { useState } from 'react';
+import { FunctionComponent } from 'react';
+import TextField from '@material-ui/core/TextField';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import Button from '@material-ui/core/Button';
+import { Formik, FormikHelpers } from 'formik';
+import * as Yup from 'yup';
+import Box from '@material-ui/core/Box';
+import OutlinedInput from '@material-ui/core/OutlinedInput';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import DateFnsUtils from '@date-io/date-fns';
+import { MuiPickersUtilsProvider, KeyboardDatePicker } from '@material-ui/pickers';
+import Grid from '@material-ui/core/Grid';
+import { CircularProgress } from '@material-ui/core';
+import PreferedDesigns from '../PreferedDesigns/PreferedDesigns';
+
+import useStyles from './useStyles';
+
+interface Props {
+  handleSubmit: (
+    {
+      title,
+      description,
+      prizeAmount,
+      date,
+      time,
+      timezone,
+    }: {
+      title: string;
+      description: string;
+      prizeAmount: string;
+      date: string;
+      time: string;
+      timezone: string;
+    },
+    {
+      setSubmitting,
+    }: FormikHelpers<{
+      title: string;
+      description: string;
+      prizeAmount: string;
+      date: string;
+      time: string;
+      timezone: string;
+    }>,
+  ) => void;
+}
+
+const NewContestForm: FunctionComponent<Props> = ({ handleSubmit }: Props): JSX.Element => {
+  const classes = useStyles();
+  const [selectedDate, setSelectedDate] = useState<Date | null>(new Date());
+
+  const handleDateChange = (date: Date | null) => {
+    setSelectedDate(date);
+  };
+
+  const times = [
```

Comment:
Maybe consider using a Time Picker, it will make things a lot easier for you :) 

Ref: https://material-ui.com/components/pickers/

Reply:
I looked at it and really hated the giant clock.  :)